core/modules/filters.js

```diff
@@ -161,6 +162,20 @@ exports.filterTiddlers = function(filterString,widget,source) {
 };
 
 /*
+Evaluates a condition and adds an item to the result-set of a filter operation depending on negation
+	title: tiddler title
+	results: the results array
+	operator: the filter operator
+	condition: the condition depending on which to add to the results (boolean)
+Returns boolean, whether or not to add an item
+*/
+exports.addResult = function(title,results,operator,condition) {
```

Comment:
I'm actually not so sure about this function anymore as you were implying it would be less performing as compared to having the negation conditional outside and then two iterator function blocks within... which would require to make sure for every single filter that we're properly negating... if that's even a concern anymore.